bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -686,21 +688,25 @@ int runCmd(CommandLine cmdLine) throws Exception {
                 predicate = replicasList -> !replicasList.contains(excludingBookieId);
             }
 
-            ZooKeeper zk = null;
+            LedgerManagerFactory mFactory = null;
             try {
-                zk = ZooKeeperClient.newBuilder()
-                        .connectString(bkConf.getZkServers())
-                        .sessionTimeoutMs(bkConf.getZkTimeout())
-                        .build();
-                LedgerManagerFactory mFactory = LedgerManagerFactory.newLedgerManagerFactory(bkConf, zk);
+                mFactory = LedgerManagerFactory.newLedgerManagerFactory(
+                    bkConf,
+                    new ZkLayoutManager(
```

Comment:
I think it would be better to avoid construct ZkLayoutManager directly.

A better way to do this:

```
try (RegistrationManager rm = ...) {
    try (LedgerManagerFactory mFactory = LedgerManagerFactory.newLedgerManagerFactory(.., rm.getLayoutManager()) {
       ...
    }
}
```

Reply:
Thanks, will try it.